cmd/client_test.go

```diff
@@ -111,3 +111,78 @@ func TestClient_CreateJobSpec(t *testing.T) {
 		assert.Equal(t, test.nJobs, len(numberOfJobs))
 	}
 }
+
+func TestClient_CreateJobRun_Success(t *testing.T) {
+	t.Parallel()
+	app, cleanup := cltest.NewApplication()
+	defer cleanup()
+	client, _ := cltest.NewClientAndRenderer(app.Store.Config)
+
+	j := cltest.NewJobWithWebInitiator()
+	assert.Nil(t, app.Store.SaveJob(&j))
+
+	set := flag.NewFlagSet("run", 0)
+	set.Parse([]string{j.ID, `{"value": 100}`})
+	c := cli.NewContext(nil, set, nil)
+	assert.Nil(t, client.CreateJobRun(c))
+}
+
+func TestClient_CreateJobRun_EmptyBody(t *testing.T) {
+	t.Parallel()
+	app, cleanup := cltest.NewApplication()
+	defer cleanup()
+	client, _ := cltest.NewClientAndRenderer(app.Store.Config)
+
+	j := cltest.NewJobWithWebInitiator()
+	assert.Nil(t, app.Store.SaveJob(&j))
+
+	set := flag.NewFlagSet("run", 0)
+	set.Parse([]string{j.ID})
+	c := cli.NewContext(nil, set, nil)
+	assert.Nil(t, client.CreateJobRun(c))
+}
+
+func TestClient_CreateJobRun_Create_InvalidBody(t *testing.T) {
+	t.Parallel()
+	app, cleanup := cltest.NewApplication()
+	defer cleanup()
+	client, _ := cltest.NewClientAndRenderer(app.Store.Config)
+
+	j := cltest.NewJobWithWebInitiator()
+	assert.Nil(t, app.Store.SaveJob(&j))
+
+	set := flag.NewFlagSet("run", 0)
+	set.Parse([]string{j.ID, "{"})
+	c := cli.NewContext(nil, set, nil)
+	assert.NotNil(t, client.CreateJobRun(c))
+}
+
+func TestClient_CreateJobRun_NotFound(t *testing.T) {
+	t.Parallel()
+	app, cleanup := cltest.NewApplication()
+	defer cleanup()
+	client, _ := cltest.NewClientAndRenderer(app.Store.Config)
+
+	j := cltest.NewJobWithWebInitiator()
+	assert.Nil(t, app.Store.SaveJob(&j))
+
+	set := flag.NewFlagSet("run", 0)
+	set.Parse([]string{"badID"})
+	c := cli.NewContext(nil, set, nil)
+	assert.NotNil(t, client.CreateJobRun(c))
+}
+
+func TestClient_CreateJobRun_WithoutWebInitiator(t *testing.T) {
+	t.Parallel()
```

Comment:
Could you collapse these 5 tests into one table driven test?

Reply:
sure thing! :+1: thanks for the feedback
do you know why the build took a long time?

Reply:
I actually don't know. I'll look into it. Rebasing master and force pushing will start a new build that shouldn't take forever.

Reply:
Ok sounds good

Reply:
ahh it wasnt building because i needed to rebase with upstream :+1: